content/200-concepts/100-components/02-prisma-client/060-full-text-search.mdx

```diff
@@ -59,41 +79,137 @@ const result = await prisma.posts.findMany({
+| `+dog +cat` | No     | The text contains 'dog' but not 'cat'   |
+| `-cat`      | Yes    | 'cat' is not in the text                |
+| `fox dog`   | Yes    | The text contains 'fox' or 'cat'        |
+| `-cat -pig` | No     | The text doesn't contain 'cat' or 'pig' |
+
+For the full range of supported operations, see the [MySQL full text search documentation](https://dev.mysql.com/doc/refman/8.0/en/fulltext-boolean.html).
+
+## Adding indexes
 
-## Adding Indexes
+### PostgreSQL
 
-To speed up your full-text queries, you should add an index to your database.
-Prisma Migrate currently doesn't support adding search indices, but you can
-easily add one yourself.
+To speed up your full-text queries, you should add an index to your database. Prisma Migrate currently doesn't support adding search indices in PostgreSQL, so this should be added in SQL. For example, the following SQL statement would add an index called `post_body_index` on the `posts_body_index` column of the `posts` table:
 
 ```sql
 CREATE INDEX post_body_index ON posts USING GIN (body);
 ```
 
-You can continue using Prisma Migrate as you were before, it will ignore indexes
-that it doesn't know about.
+For further information, see the [Postgres documentation on indexes](https://www.postgresql.org/docs/12/sql-createindex.html).
+
+You can continue using Prisma Migrate as you were before, it will ignore indexes that it doesn't know about.
+
+### MySQL
+
+For MySQL, it is necessary to add indexes to any columns you search using the `@@fulltext` argument in the `schema.prisma` file. To do this, the `"fullTextIndex"` preview feature must be enabled.
+
+In the following example, one full text index is added to the `content` field of the `Blog` model, and another is added to both the `content` and `title` fields together:
+
+```prisma file=schema.prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["fullTextSearch", "fullTextIndex"]
+}
+
+model Blog {
+  id      Int    @unique
+  content String
+  title   String
 
-## Limitations
+  @@fulltext([content])
+  @@fulltext([content, title])
+}
+```
+
+The first index allows searching the `content` field for occurrences of the word 'cat':
+
+```prisma file=schema.prisma
+const result = await prisma.blogs.findMany({
+where: {
+content: {
+search: 'cat',
+},
+},
+})
+```
+
+The second index allows searching both the `content` and `title` fields for occurrences of the word 'cat' in the `content` and 'food' in the `title`:
+
+```prisma file=schema.prisma
+const result = await prisma.blogs.findMany({
+where: {
+content: {
+search: 'cat',
+},
+title: {
+search: 'food'
+}
+},
+})
+```
 
-- The `search` field is only supported in PostgreSQL
-- You'll need to manage indexes yourself.
+However, if you try to search on `title` alone, the search will fail with the error "Cannot find a fulltext index to use for the search and the message code is P2030", because the index requires a search on both fields.
```

Comment:
Wasn't sure whether to include all this detail about the error message, but putting it in for now

Reply:
```suggestion
However, if you try to search on `title` alone, the search will fail with the error "Cannot find a fulltext index to use for the search" and the message code is P2030, because the index requires a search on both fields.
```